lib/logstash/filters/cipher.rb

```diff
@@ -157,11 +157,15 @@ def filter(event)
         encryptedKey = Base64.strict_decode64(dataHash["encryptedKey"])
         encryptionContext = eval(dataHash['encryptionContext'])
 
-        decryptedKey = @kms.decrypt(:ciphertext_blob =>  encryptedKey,
-                                   encryption_context: encryptionContext
-                                  )
+        unless @keys_hash.has_key? encryptedKey
+          @logger.warn("YES FETCH")
+          @decryptedKey = @kms.decrypt(:ciphertext_blob =>  encryptedKey,
```

Comment:
Also, I don't think decryptedKey needs an @, it doesn't look like it's used outside of this method